substra/sdk/backends/local/backend.py

```diff
@@ -739,8 +736,309 @@ def describe(self, asset_type, key):
+            },
+            'creator': testtuple.creator,
+            'key': testtuple.key,
+            'perf': testtuple.dataset.perf,
+            'tag': testtuple.tag,
+            'traintupleKey': testtuple.traintuple_key
+        }
+
+    def leaderboard(self, objective_key, sort='desc'):
+        objective = self._db.get(schemas.Type.Objective, objective_key)
+        testtuples = self._db.list(schemas.Type.Testtuple)
+        certified_testtuples = [
+            self.__format_for_leaderboard(t)
+            for t in testtuples
+            if t.objective_key == objective_key and t.certified
+        ]
+        certified_testtuples.sort(key=lambda x: x['perf'], reverse=(sort == 'desc'))
+        board = {
+            'objective': objective.to_response(),
+            'testtuples': certified_testtuples
+        }
+        return board
 
     def cancel_compute_plan(self, compute_plan_id):
+        # Execution is synchronous in the local backend so this
+        # function does not make sense.
         raise NotImplementedError
+
+    def update_compute_plan(self, compute_plan_id: str, spec: schemas.UpdateComputePlanSpec):
+        compute_plan = self._db.get(schemas.Type.ComputePlan, compute_plan_id)
+
+        # Get all the new tuples and their dependencies
+        (
+            all_tuples,
+            traintuples,
+            aggregatetuples,
+            compositetuples
+        ) = self.__get_all_tuples_compute_plan(spec)
+
+        # Define the rank of each traintuple, aggregate tuple and composite tuple
+        old_tuples = {id_: list() for id_ in compute_plan.id_to_key}
+        all_tuples.update(old_tuples)
+
+        # Get the rank of all the tuples already in the compute plan
+        visited = dict()
+        if compute_plan.traintuple_keys:
+            for key in compute_plan.traintuple_keys:
+                id_, rank = self.__get_id_rank_in_compute_plan(
+                    schemas.Type.Traintuple,
+                    key,
+                    compute_plan.id_to_key
+                )
+                visited[id_] = rank
+
+        if compute_plan.aggregatetuple_keys:
+            for key in compute_plan.aggregatetuple_keys:
+                id_, rank = self.__get_id_rank_in_compute_plan(
+                    schemas.Type.Traintuple,
+                    key,
+                    compute_plan.id_to_key
+                )
+                visited[id_] = rank
+
+        if compute_plan.composite_traintuple_keys:
+            for key in compute_plan.composite_traintuple_keys:
+                id_, rank = self.__get_id_rank_in_compute_plan(
+                    schemas.Type.Traintuple,
+                    key,
+                    compute_plan.id_to_key
+                )
+                visited[id_] = rank
+
+        while len(visited) != len(all_tuples):
+            node = set(all_tuples.keys()).difference(set(visited.keys())).pop()
+            self.__get_rank(node, visited, set(), all_tuples)
```

Comment:
I like this very much, it could also serve as the basis for the future autobatching feature for compute plan creation on remote backends. Do you think it could be moved to an independent function with dedicated tests?

Reply:
Yes, sure ! The function will contain all of this code:
```python
        while len(visited) != len(all_tuples):
            node = set(all_tuples.keys()).difference(set(visited.keys())).pop()
            self.__get_rank(node, visited, set(), all_tuples)
```
is that right ?

Reply:
Yes, absolutely.

Reply:
I put it in assets.py (and realised that the rest of the file was useless (did a search on all repos)), I do not know if it is the best place :/

Reply:
Thank you Travis, assets was used to generate cli.md😅